### PR TITLE
[release-v1.40] Automated cherry pick of #5482: Set ingress DNSRecord component values even if the addon is disabled

### PR DESCRIPTION
**Pull request by ialidzhikov**
/kind/bug
/area/quality

Cherry pick of #5482 on release-v1.40.

#5482: Set ingress DNSRecord component values even if the addon is disabled

**Release Notes:**
```bugfix operator
An issue preventing the nginx-ingress addon to be disabled is now fixed.
```

### CODE REVIEW

**Comment by ialidzhikov** [comment]
/hold
until the parent PR is merged

**Review by rfranzke — APPROVED** [review]

**Pull request merged**
